plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Swap.hs

```diff
@@ -59,13 +62,21 @@ data SwapOwners = SwapOwners {
 PlutusTx.makeIsData ''SwapOwners
 PlutusTx.makeLift ''SwapOwners
 
-type SwapOracle = OracleValue Rational
+type SwapOracleMessage = SignedMessage (Observation Rational)
 
-mkValidator :: Swap -> SwapOwners -> SwapOracle -> PendingTx -> Bool
+mkValidator :: Swap -> SwapOwners -> SwapOracleMessage -> PendingTx -> Bool
```

Comment:
Unrelated, but it looks like we don't have any tests for `Swap` (I was looking for `swap.pir`...)

Reply:
Yeah when I first wrote it there were several limitations (no multi currency, no working oracles, no rational numbers, etc) so I didn't bother with the client code. It probably makes sense to re-write it in the style of the futures contract now.